File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

void print_repeated(string s, int n);
void print_spaces(int n);
void print_blocks(int n);

int main(void)
{
    // promt user, continue until a valid value is given
    int height;
    string prompt = "Supply a positive integer no larger than 23: ";
    do
    {
        height = get_int("%s\n", prompt);
    }
    while (height < 0 || height > 23);

    // loop to build structure
    int steps = 1;
    while (steps <= height)
    {
        print_spaces(height - steps);
        print_blocks(steps);
        print_spaces(2);
        print_blocks(steps);
        printf("\n");
        steps++;
    }
}

// print a string multiple times
void print_repeated(string s, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%s", s);
    }
}

// print n spaces
void print_spaces(int n)
{
    print_repeated(" ", n);
}

// print n blocks
void print_blocks(int n)
{
    print_repeated("#", n);
}

File: pset6/mario/mario.py
# prints a pyramid of hashtags based on user input

from cs50 import get_int

def main():
    while True:
        height = get_int("Supply a positive integer no larger than 23: ")
        if height >= 0 and height <= 23:
            break
    for steps in range(1, height + 1):
        print(" " * (height - steps), end="")
        print("#" * steps, end="")
        print("  ", end="")
        print("#" * steps)

if __name__ == "__main__":
    main()